pkg/cmd/admin/top/graph.go

```diff
@@ -0,0 +1,149 @@
+				topLayerAdded = true
+			}
+			g.AddEdge(imageNode, layerNode, edgeKind)
+			glog.V(4).Infof("Adding image layer %q to graph (%q) ", layer.Name, edgeKind)
+		}
+	}
+}
+
+func addImageStreamsToGraph(g graph.Graph, streams *imageapi.ImageStreamList) {
+	for i := range streams.Items {
+		stream := &streams.Items[i]
+		glog.V(4).Infof("Adding ImageStream %s/%s to graph", stream.Namespace, stream.Name)
+		isNode := imagegraph.EnsureImageStreamNode(g, stream)
+		imageStreamNode := isNode.(*imagegraph.ImageStreamNode)
+
+		// connect IS with underlying images
+		for tag, history := range stream.Status.Tags {
+			for i := range history.Items {
+				image := history.Items[i]
+				n := imagegraph.FindImage(g, image.Image)
+				if n == nil {
+					glog.V(2).Infof("Unable to find image %q in graph (from tag=%q, dockerImageReference=%s)",
+						history.Items[i].Image, tag, image.DockerImageReference)
+					continue
+				}
+				imageNode := n.(*imagegraph.ImageNode)
+				glog.V(4).Infof("Adding edge from %q to %q", imageStreamNode.UniqueName(), imageNode.UniqueName())
+				edgeKind := ImageStreamEdgeKind
+				if i > 1 {
+					edgeKind = HistoricImageStreamEdgeKind
+				}
+				g.AddEdge(imageStreamNode, imageNode, edgeKind)
+			}
+		}
+	}
+}
+
+func addPodsToGraph(g graph.Graph, pods *kapi.PodList) {
+	for i := range pods.Items {
+		pod := &pods.Items[i]
+		if pod.Status.Phase != kapi.PodRunning && pod.Status.Phase != kapi.PodPending {
+			glog.V(4).Infof("Pod %s/%s is not running nor pending - skipping", pod.Namespace, pod.Name)
+			continue
+		}
+
+		glog.V(4).Infof("Adding pod %s/%s to graph", pod.Namespace, pod.Name)
+		podNode := kubegraph.EnsurePodNode(g, pod)
+		addPodSpecToGraph(g, &pod.Spec, podNode)
+	}
+}
+
+func addPodSpecToGraph(g graph.Graph, spec *kapi.PodSpec, predecessor gonum.Node) {
+	for j := range spec.Containers {
+		container := spec.Containers[j]
+
+		glog.V(4).Infof("Examining container image %q", container.Image)
+		ref, err := imageapi.ParseDockerImageReference(container.Image)
+		if err != nil {
+			glog.V(2).Infof("Unable to parse DockerImageReference %q: %v - skipping", container.Image, err)
+			continue
+		}
+
+		if len(ref.ID) == 0 {
+			// ignore not managed images
+			continue
+		}
+
+		imageNode := imagegraph.FindImage(g, ref.ID)
+		if imageNode == nil {
+			glog.V(1).Infof("Unable to find image %q in the graph", ref.ID)
+			continue
+		}
+
+		glog.V(4).Infof("Adding edge from %v to %v", predecessor, imageNode)
+		g.AddEdge(predecessor, imageNode, PodImageEdgeKind)
+	}
+}
+
+func markParentsInGraph(g graph.Graph) {
+	imageNodes := getImageNodes(g.Nodes())
+	for _, in := range imageNodes {
+		// find image's top layer, should be just one
+		for _, e := range g.OutboundEdges(in, ImageTopLayerEdgeKind) {
+			layerNode, _ := e.To().(*imagegraph.ImageLayerNode)
+			// find image's containing this layer but not being their top layer
+			for _, ed := range g.InboundEdges(layerNode, ImageLayerEdgeKind) {
```

Comment:
It's reasonable, but it'll be a very rare cases. Can you open a separate issue, I'll address that as a followup, if you don't mind?